### ISSUE TITLE
Unable to mount a EFS volume to a pod in EKS

### ISSUE THREAD

**Issue by minhyeok-kweun** [aws/efs-utils#72]
I'm using a EFS volume as a shared volume across pods on EKS. up until yesterday, I had no problem in mounting the volume to pods. But since yesterday I have been unable to mount it with the error log below:
I setup the volume according to [this guide](https://aws.amazon.com/ko/premiumsupport/knowledge-center/eks-persistent-storage/). I think there is nothing wrong with my setup and configuration, since it worked so far. But all of sudden it just failed and I've searched on the internet but couldn't just find any clue to resolve this issue.

Is any body there who can help me fix this?
``
MountVolume.SetUp failed for volume "fsroot-pv" : kubernetes.io/csi: mounter.SetupAt failed: rpc error: code = Internal desc = Could not mount "fs-60f3c401:/" at "/var/lib/kubelet/pods/997a1ffa-8729-484c-9a89-a3c07be63cb8/volumes/kubernetes.io~csi/fsroot-pv/mount": mount failed: exit status 1 Mounting command: mount Mounting arguments: -t efs fs-60f3c401:/ /var/lib/kubelet/pods/997a1ffa-8729-484c-9a89-a3c07be63cb8/volumes/kubernetes.io~csi/fsroot-pv/mount Output: Traceback (most recent call last): File "/sbin/mount.efs", line 1375, in <module> main() File "/sbin/mount.efs", line 1355, in main bootstrap_logging(config) File "/sbin/mount.efs", line 1031, in bootstrap_logging raw_level = config.get(CONFIG_SECTION, 'logging_level') File "/lib64/python2.7/ConfigParser.py", line 607, in get raise NoSectionError(section) ConfigParser.NoSectionError: No section: 'mount'
``

**Comment by minhyeok-kweun** [comment]
I don't even know what kinds of information I should provide to debug this. And it seems it doesn't have anything to do with my configuration..

**Comment by Cappuccinuo** [comment]
Hey @minhyeok-kweun ,

Is there a mount section in the config file that csi driver uses? 

https://github.com/aws/efs-utils/blob/master/dist/efs-utils.conf#L16

**Comment by kbasv** [comment]
Hey @minhyeok-kweun 

I was able to reproduce on my efs-csi-driver pod by delete [mount] from efs-utils.conf.
Can you provide us the contents efs-utils.conf on your efs-csi-driver pod which is located under the following section: /etc/amazon/efs/efs-utils.conf

**Comment by minhyeok-kweun** [comment]
@Cappuccinuo @kbasv 
Hmm I couldn't find a pod named 'efs-csi-driver', instead I checked /etc/amazon/efs/ on a 'efs-csi-node' pod. But I couldn't find any file there, it was just empty.

**Comment by minhyeok-kweun** [comment]
As far as I understand, it seems the directory /etc/amazon/efs/ is mounted as host path to a coresponding dir on the ec2 instance it's running on. I guess the configuration file had to be the directory on the instance, before the path was being mounted. Or does efs-util create one if there is no config file at startup?

**Comment by minhyeok-kweun** [comment]
Here is log massage I found from efs-csi-node container. The first couple of lines are suspicious to be related to this issue.

```
I0618 23:37:36.298958       1 mount_linux.go:163] Cannot run systemd-run, assuming non-systemd OS
I0618 23:37:36.299055       1 mount_linux.go:164] systemd-run failed with: exit status 1
I0618 23:37:36.299103       1 mount_linux.go:165] systemd-run output: Failed to create bus connection: No such file or directory
I0618 23:37:36.299325       1 driver.go:86] Starting watchdog
I0618 23:37:36.299460       1 driver.go:90] Staring subreaper
I0618 23:37:36.299477       1 driver.go:93] Listening for connections on address: &net.UnixAddr{Name:"/csi/csi.sock", Net:"unix"}
E0618 23:37:36.344812       1 efs_watch_dog.go:131] amazon-efs-mount-watchdog[Error]: Traceback (most recent call last):
  File "/usr/bin/amazon-efs-mount-watchdog", line 948, in <module>
E0618 23:37:36.348788       1 efs_watch_dog.go:85] Process amazon-efs-mount-watchdog exits exit status 1
W0618 23:37:36.348807       1 reaper.go:59] Failed to wait for child process no child processes
E0618 23:37:36.391115       1 efs_watch_dog.go:131] amazon-efs-mount-watchdog[Error]: Traceback (most recent call last):
  File "/usr/bin/amazon-efs-mount-watchdog", line 948, in <module>
E0618 23:37:36.395474       1 efs_watch_dog.go:85] Process amazon-efs-mount-watchdog exits exit status 1
```

**Comment by minhyeok-kweun** [comment]
When I installed the driver via Helm using the command below, it seems it's working now.
I guess there may be some difference between manual installation and using Helm.
It could be a issue in [aws-efs-csi-driver](https://github.com/kubernetes-sigs/aws-efs-csi-driver). I'll raise an issue there if needed.
```
helm install aws-efs-csi-driver "https://github.com/kubernetes-sigs/aws-efs-csi-driver/releases/download/v0.3.0/helm-chart.tgz"
```

**Comment by jqmichael** [comment]
What is the driver version you installed when seeing the bug? 

Was that from v0.3.0 or tip of master?

**Comment by jqmichael** [comment]
Tracking the fix at https://github.com/kubernetes-sigs/aws-efs-csi-driver/issues/191

**Comment by youwalther65** [comment]
Had the same issue starting yesterday. The problem was solved for me by instaling the efs-utils on all theworker nodes! Maybe the underlying worker node AMI got changed wrongly! Running in eu-west-1

# yum -y install amazon-efs-utils

**Comment by youwalther65** [comment]
> 
> 
> Had the same issue starting yesterday. The problem was solved for me by instaling the efs-utils on all theworker nodes! Maybe the underlying worker node AMI got changed wrongly! Running in eu-west-1
> # yum -y install amazon-efs-utils

The volume is just mounted into the efs-csi-node pods efs-plugin containers running in kube-system namespace. See volume section of EFS CSI node definition:

 efs-utils-config:
    Type:          HostPath (bare host directory volume)
    Path:          /etc/amazon/efs
    HostPathType:  DirectoryOrCreate

If the host path is empty nothing is available in the CSI containers! So for me a clear AMI issue! I used eksctl last version

**Comment by keifgwinn** [comment]
I'm running into the same thing on newly spun up nodes in the cluster. 

eu-west-2 : amazon-eks-node-1.16-v20200531 (ami-0c347c2a3a6cfc4b7)

**Comment by kbasv** [comment]
@keifgwinn Can you provide the following:

1. How did you setup the efs-csi-driver?
2. What is the version of efs-csi-driver?
3. Logs from efs-csi-driver and mount.log ?

**Comment by keifgwinn** [comment]
I followed the instructions here https://docs.aws.amazon.com/eks/latest/userguide/efs-csi.html 35 days ago, but reading this thread and seeing the new 1.0.0 release I've upgrade to the helm chart and it's working now

**Comment by kbasv** [comment]
Issue is now resolved with 1.0.0 release of efs-csi-driver. Closing this issue.
